src/coord/rmr/redis_cluster.c

```diff
@@ -130,13 +130,23 @@ void UpdateTopology(RedisModuleCtx *ctx) {
 }
 
 #define REFRESH_PERIOD 1000 // 1 second
+RedisModuleTimerID topologyRefreshTimer = 0;
 
 static void UpdateTopology_Periodic(RedisModuleCtx *ctx, void *p) {
   REDISMODULE_NOT_USED(p);
-  RedisModule_CreateTimer(ctx, REFRESH_PERIOD, UpdateTopology_Periodic, NULL);
+  topologyRefreshTimer = RedisModule_CreateTimer(ctx, REFRESH_PERIOD, UpdateTopology_Periodic, NULL);
```

Comment:
Do you know what is the reason that the timer is created BEFORE `UpdateTopology` (if any)?

Reply:
I know because I intentionally wrote it this way
From the [docs](https://redis.io/docs/latest/develop/reference/modules/modules-api-ref/#redismodule_createtimer):

> Note that for the common use case of a repeating timer (Re-registration of the timer inside the RedisModuleTimerProc callback), it matters when this API is called: If it is called at the beginning of 'callback' it means the event will be triggered every 'period'. If it is called at the end of 'callback' it means there will be 'period' milliseconds gaps between events.